src/windows/gfx/d3d11-ctx.c

```diff
@@ -242,6 +435,53 @@ static void d3d11_ctx_refresh(struct d3d11_ctx *ctx)
 			d3d11_ctx_init(ctx);
 		}
 	}
+
+	if (ctx->hdr != ctx->hdr_new) {
+		// If in HDR mode, we keep swap chain in HDR10 (rec2020 10-bit RGB + ST2084 PQ);
+		// otherwise in SDR mode, it's the standard BGRA8 sRGB
+		DXGI_FORMAT format = ctx->hdr_new ? DXGI_FORMAT_R10G10B10A2_UNORM : DXGI_FORMAT_B8G8R8A8_UNORM;
+		DXGI_COLOR_SPACE_TYPE colorspace = ctx->hdr_new ? DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020 : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+
+		HRESULT e = IDXGISwapChain2_ResizeBuffers(ctx->swap_chain2, 0, 0, 0, format, D3D11_SWFLAGS);
+		if (e == S_OK) {
+			e = IDXGISwapChain3_SetColorSpace1(ctx->swap_chain3, colorspace);
+
+			if (e == S_OK) {
+				ctx->hdr = ctx->hdr_new;
```

Comment:
How often `d3d11_ctx_refresh` is called? Let's say `ResizeBuffers` or `SetColorSpace1` continuously fail, this part will be called indefinitely and the context will be re-created over and over, is it intended?

Reply:
`d3d11_ctx_refresh` is called every frame. Good catch. We do want to keep trying again because there could be transient reasons for failure. But we also don't want to keep re-creating forever. Good catch. I will limit the number of tries and if it still fails, just never use the path at all.